src/System.IO.FileSystem/tests/FileInfo/Length.cs

```diff
@@ -39,5 +39,25 @@ public void Length_Of_Directory_Throws_FileNotFoundException()
             FileInfo info = new FileInfo(path);
             Assert.Throws<FileNotFoundException>(() => info.Length);
         }
+
+        [SkippableFact]
+        public void SymLinkLength()
+        {
+            var path = GetTestFilePath();
+            var linkPath = GetTestFilePath();
+            CreateNonEmptyFile(path, 500);
+            // In some cases (such as when running without elevated privileges,
+            // the symbolic link may fail to create. Only run this test if it creates
+            // correctly.
+            Skip.IfNot(MountHelper.CreateSymbolicLink(linkPath, path));
+
+            var info = new FileInfo(path);
+            var linkInfo = new FileInfo(linkPath);
+            Assert.Equal(500, info.Length);
+            // On Windows, sym links report 0 size; on Linux, their size is the length of the path they point to
+            // Confirm that the size we get is not the size of the target file (and that it's less, since our temporary
+            // sym links should never exceed 500 bytes.
```

Comment:
Some of the paths in our CI system can get pretty long... I wouldn't be surprised if they bump up against this at some point.  You might choose a larger value.